state_machine/applications/system-check/main.py

```diff
@@ -0,0 +1,115 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+* Author(s): Yashika Batra
+"""
+
+from lib.pycubed import pocketqube as cubesat
+import tests
+import tests.imu_test
+import tests.radio_test
+import tests.sun_sensor_test
+import tests.coil_test
+import tests.burnwire_test
+
+"""
+TODO:
+run through files and documents for last check
+fix string formatting
+"""
+
+# initialize hardware_dict and result_dict
+hardware_dict = cubesat.hardware
+result_dict = {
+    'SD_Card_Logging': ('', False),
+    'IMU_Acc_Still': ('', False),
+    'IMU_Acc_Moving': ('', False),
+    'IMU_Gyro_Still': ('', False),
+    'IMU_Gyro_Turntable': ('', False),
+    'IMU_Mag_Standalone': ('', False),
+    'IMU_Mag_Magnet': ('', False),
+    'IMU_Temp': ('', False),
+    'Radio_Receive_Beacon': ('', False),
+    'Radio_Send_Beacon': ('', False),
+    'Sun_MinusY_Dark': ('', False),
+    'Sun_MinusY_Light': ('', False),
+    'Sun_MinusZ_Dark': ('', False),
+    'Sun_MinusZ_Light': ('', False),
+    'Sun_MinusX_Dark': ('', False),
+    'Sun_MinusX_Light': ('', False),
+    'Sun_PlusY_Dark': ('', False),
+    'Sun_PlusY_Light': ('', False),
+    'Sun_PlusZ_Dark': ('', False),
+    'Sun_PlusZ_Light': ('', False),
+    'Sun_PlusX_Dark': ('', False),
+    'Sun_PlusX_Light': ('', False),
+    'CoilDriverX_VoltTest': ('', False),
+    'CoilDriverY_VoltTest': ('', False),
+    'CoilDriverZ_VoltTest': ('', False),
+    'Burnwire1_Volt1': ('', False),
+    'Burnwire1_Volt2': ('', False),
+    'Burnwire1_Volt3': ('', False),
+    'Burnwire2_Volt1': ('', False),
+    'Burnwire2_Volt2': ('', False),
+    'Burnwire2_Volt3': ('', False)
+}
+
+# print acknowledgement that test has started
+print("Running System Check...")
+print("Initialization has concluded. Printing results...")
+print("")
+for entry in hardware_dict:
+    print(str(entry.key), str(entry.val))
+print("")
+
+# check that solar boards are attached
+run_sun_sensor = False
+run_coil_driver = False
+run_burnwire = False
+solar_boards = input("Are the solar boards attached to the mainboard? (Y/N): ")
```

Comment:
I don't think we actually need to ask this question. I'd rather have the code try to read the sun sensors, etc, even if there are no solar boards attached.

Reply:
Then report that the sun sensors failed the test.

Reply:
It checks if the sun sensors are attached to the board / tries to read them earlier in the check. This check is built into the pycubed library, so when we initialize a satellite object we also initialize a hardware dictionary, which is what is printed earlier on in this file.

The input being asked for here feeds into a secondary hardware check, which will check for the functionality of the sun sensors, coil drivers, and burnwire ICs (which are all on the solar boards)

I can still remove this if you'd like though, as each individual file checks that the hardware exists before running the second-layer functionality test, and updates the results dictionary accordingly. All the results are printed out later in the test anyways, so it shouldn't be an issue.